src/ucp/core/ucp_context.c

```diff
@@ -28,19 +28,26 @@ static const char *ucp_atomic_modes[] = {
     [UCP_ATOMIC_MODE_LAST]   = NULL,
 };
 
+static const char * ucp_device_type_names[] = {
+    [UCT_DEVICE_TYPE_NET]  = "network",
+    [UCT_DEVICE_TYPE_SHM]  = "intra-node",
+    [UCT_DEVICE_TYPE_ACC]  = "accelerator",
+    [UCT_DEVICE_TYPE_SELF] = "loopback",
+};
```

Comment:
So besides already existing name you are trying to add the dimension   of type ?  Does it mean that the uses has to specify the both ?